images/providers.go

```diff
@@ -0,0 +1,100 @@
+package images
+
+import (
+	"context"
+	"errors"
+
+	"github.com/grafana/alerting/logging"
+	"github.com/grafana/alerting/models"
+	"github.com/prometheus/alertmanager/types"
+)
+
+// ErrImageUploadNotSupported is returned when image uploading is not supported.
+var ErrImageUploadNotSupported = errors.New("image upload is not supported")
```

Comment:
The comment is a bit redundant. I'd either add more details (e.g. why this can happen) or remove it.

Suggestion: I just know we're going to forget what this error means and where it comes from. We could make it more explicit so if we find it in the logs, we know right away what this is about.
```suggestion
// ErrImageUploadNotSupported is returned when the image is provided as a URL in an annotation,
// as these URLs cannot be trusted for direct download/upload.
var ErrImageUploadNotSupported = errors.New("image upload is not supported when using URLs in annotations")
```

Reply:
Good point, but I'd like the keep the error generic. I'll add context when used (similar to how we do validation errors). Ex:
```go
fmt.Errorf("%w: URLProvider does not support raw image data", ErrImageUploadNotSupported)
```